mmstudio/src/main/java/org/micromanager/display/inspector/internal/panels/intensity/IntensityInspectorPanelController.java

```diff
@@ -587,18 +587,22 @@ public void onEvent(DisplaySettingsChangedEvent e) {
 
    @Subscribe
    public void onEvent(DataProviderHasNewImageEvent event) {
-      final int channel = event.getImage().getCoords().getChannel();
-      SwingUtilities.invokeLater(() -> {
-         try {
-            if (channel >= channelControllers_.size()) {
-               setUpChannelHistogramsPanel(channel + 1);
+      // (NS - 2020-03-27)
+      // Hack: handle only if the circular buffer is not too full.  How full is highly arbitrary!
+      if ( !studio_.acquisitions().isAcquisitionRunning() || studio_.core().getRemainingImageCount() < 6 ) {
```

Comment:
As we discussed on the phone I'm very much for using the sequence buffer fill level to regulate display rate. When things are working smoothly the buffer stays at 0-2 images most of the time, so 6 seems reasonable until proven otherwise.

I think there is one set of edge cases where this will cause problems: MDAs can also be run with snaps only (making `isAcquisitionRunning` return `true` even if no sequence acquisition is running). Yet a previous sequence acquisition could have left more than 5 images in the sequence buffer (especially if the previous sequence acquisition was Live).

Reply:
Great catch!  I added a check to isSequenceRunning, and at least the edge case you describe is now working.  I hope that there are not more....

Reply:
Well, I think that covers most cases. Because of the time delay between acquisition and this code, an MDA that mixes snap and sequence acquisition might have strange display behavior, but that's a harder one to fix. 

The other issue that I would worry about is that the last image(s) of an acquisition may never reach the display -- but I think you would have seen that in your testing, so I'm assuming some other mechanism is covering that case?